java/java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertTextBlockToString.java

```diff
@@ -0,0 +1,106 @@
+import org.openide.util.NbBundle.Messages;
+
+@Hint(displayName = "#DN_ConvertTextBlockToString", description = "#DESC_ConvertTextBlockToString", category = "general", minSourceVersion = "13")
+@Messages({
+    "DN_ConvertTextBlockToString=Convert Text block to String",
+    "DESC_ConvertTextBlockToString=Text Block may not be supported in older versions of java older then 13"
+})
+public class ConvertTextBlockToString {
+
+    @TriggerTreeKind(Tree.Kind.STRING_LITERAL)
+    @Messages("ERR_ConvertTextBlockToString=Text block may not be supported")//NOI18N
+    public static ErrorDescription computeWarning(HintContext ctx) {
+        TokenSequence<?> ts = ctx.getInfo().getTokenHierarchy().tokenSequence();
+        if(ts==null)return null;
+        ts.move((int) ctx.getInfo().getTrees().getSourcePositions().getStartPosition(ctx.getPath().getCompilationUnit(), ctx.getPath().getLeaf()));
+        if (!ts.moveNext() || ts.token().id() != JavaTokenId.MULTILINE_STRING_LITERAL) {
+            return null;
+        }
+
+        String orignalString = ctx.getPath().getLeaf().toString();
+        String orignalStringArr[] = textBlockToStringArr(orignalString.substring(1, orignalString.length() - 1));
+        Fix fix = new FixImpl(ctx.getInfo(), ctx.getPath(), orignalStringArr).toEditorFix();
+        return ErrorDescriptionFactory.forName(ctx, ctx.getPath(), Bundle.ERR_ConvertTextBlockToString(), fix);
+    }
+
+    private static String[] textBlockToStringArr(String textBlock) {
+        String oneLine[] = textBlock.split("\\\\n", -1);    // NOI18N
+        return oneLine;
+    }
+
+    private static final class FixImpl extends JavaFix {
+
+        String orignalStringArr[];
+
+        public FixImpl(CompilationInfo info, TreePath tp, String orignalStringArr[]) {
+            super(info, tp);
+            this.orignalStringArr = orignalStringArr;
+        }
+
+        @Override
+        @Messages("FIX_ConvertTextBlockToString=Convert to String")
+        protected String getText() {
+            return Bundle.FIX_ConvertTextBlockToString();
+        }
+
+        @Override
+        protected void performRewrite(TransformationContext ctx) {
+            ExpressionTree ext = buildTree(orignalStringArr, orignalStringArr.length - 1, ctx);
+            ctx.getWorkingCopy().rewrite(ctx.getPath().getLeaf(), ext);
+
+        }
+
+        static private ExpressionTree buildTree(String textBlockLines[], int currentLine, TransformationContext ctx) {
+            for (int i = 0; i < textBlockLines.length; i++) {
+                textBlockLines[i] = textBlockLines[i].replaceAll("\\Q\\\"\\E", "\""); // NOI18N
+                textBlockLines[i] = textBlockLines[i].replaceAll("\\Q\\\'\\E", "\'"); // NOI18N
+            }
+            if (textBlockLines.length == 1) {
+                return ctx.getWorkingCopy().getTreeMaker().Binary(Tree.Kind.PLUS, ctx.getWorkingCopy().getTreeMaker().Literal(textBlockLines[0]), ctx.getWorkingCopy().getTreeMaker().Literal(""));// NOI18N
```

Comment:
I don't understand this one -- if there's just line, no newlines included (ending `"""` delimiter follows the content string on the same line), the produced code is still
```
        String s = "s" + "";
```

Reply:
this is to support going back and forth for textblock to string and string to textblock.
does extra "" cause any issue??
for example if I dont put extra +"" then
          String s="""
                         akshay
                        """;
will turn into String s="akshay\n"; and this does not give hint to convert to TextBlock
However "akshay\n" + ""; does

Reply:
It does not any issue. It's just redundant so it should not be produced by a refactoring.

The question is why `String s="akshay\n";` does not allow conversion to a text block ? You can call it a feature (it IMHO actually spoils the formatting if converted, in this case) - a user how hit convert to String and does not like the outcome can Undo the operation. 

An option could be made, to provide the hint on all strings with **at least** one newline within them, or at least N newlines. The user can then tune up to his desired style.

But IMHO there's no use-case for generating redudnant code that would cause code reviewer to demand a fix.